src/plugins/displayLayout/components/TelemetryView.vue

```diff
@@ -284,14 +298,31 @@ export default {
         updateTelemetryFormat(format) {
             this.$emit('formatChanged', this.item, format);
         },
-        getContextMenuActions() {
-            let actionsObject = this.openmct.actions.get(this.currentObjectPath, {}, {viewHistoricalData: true});
-            let applicableActionKeys = Object.keys(actionsObject).filter(key => actionsObject[key].key === 'viewHistoricalData');
+        async getContextMenuActions() {
+            const defaultNotebook = getDefaultNotebook();
+
+            const domainObject = defaultNotebook && await this.openmct.objects.get(defaultNotebook.notebookMeta.identifier);
+
+            const actionsObject = this.openmct.actions.get(this.currentObjectPath, this.getViewContext(), { viewHistoricalData: true }).applicableActions;
+            let applicableActionKeys = Object.keys(actionsObject)
+                .filter(key => {
+                    const isViewHistoricalData = actionsObject[key].key === 'viewHistoricalData';
+                    const isCopyToClipboard = actionsObject[key].key === 'copyToClipboard';
+                    const isCopyToNotebook = defaultNotebook && actionsObject[key].key === 'copyToNotebook';
+
+                    if (isCopyToNotebook) {
+                        const defaultPath = domainObject && `${domainObject.name} - ${defaultNotebook.section.name} - ${defaultNotebook.page.name}`;
+                        actionsObject[key].name = `Copy to Notebook ${defaultPath}`;
+                    }
+
+                    return isCopyToClipboard || isCopyToNotebook || isViewHistoricalData;
+                });
 
-            this.contextMenuActions = applicableActionKeys.map(key => actionsObject[key]);
+            return applicableActionKeys.map(key => actionsObject[key]);
         },
-        showContextMenu(event) {
-            this.openmct.menus.showMenu(event.x, event.y, this.contextMenuActions);
+        async showContextMenu(event) {
+            const contextMenuActions = await this.getContextMenuActions();
```

Comment:
this should not be async

Reply:
Actually we need it as calling async getContextMenuActions inside this.